pkg/server/server.go

```diff
@@ -130,27 +128,19 @@ func (s *Server) Run(ctx context.Context) error {
 		// nolint:errcheck
 		go http.ListenAndServe(s.options.Extra.ProfilerAddress, nil)
 	}
+
 	if s.options.EmbeddedEtcd.Enabled {
-		es := &etcd.Server{
-			Dir: s.options.EmbeddedEtcd.Directory,
-		}
-		var listenMetricsURLs []url.URL
-		if len(s.options.EmbeddedEtcd.ListenMetricsURLs) > 0 {
-			var err error
-			listenMetricsURLs, err = etcdtypes.NewURLs(s.options.EmbeddedEtcd.ListenMetricsURLs)
-			if err != nil {
-				return err
-			}
+		config, err := embeddedetcd.NewConfig(s.options.EmbeddedEtcd)
+		if err != nil {
+			return err
 		}
-		embeddedClientInfo, err := es.Run(ctx, s.options.EmbeddedEtcd.PeerPort, s.options.EmbeddedEtcd.ClientPort, listenMetricsURLs, s.options.EmbeddedEtcd.WalSizeBytes, s.options.EmbeddedEtcd.QuotaBackendBytes, s.options.EmbeddedEtcd.ForceNewCluster, s.options.GenericControlPlane.Etcd.EnableWatchCache)
+		completed, err := config.Complete()
```

Comment:
okay, now the params are taken directly from the options.

Reply:
today the embedded etcd does not follow the options/config/server model at all.